python/ray/autoscaler/commands.py

```diff
@@ -118,24 +215,39 @@ def _bootstrap_config(config, no_config_cache=False):
             config["provider"]))
 
     provider_cls = importer(config["provider"])
-    resolved_config = provider_cls.bootstrap_config(config)
+
+    with cli_logger.timed(  # todo: better message
+            "Bootstraping {} config",
+            PROVIDER_PRETTY_NAMES.get(config["provider"]["type"])):
+        resolved_config = provider_cls.bootstrap_config(config)
+
     if not no_config_cache:
         with open(cache_key, "w") as f:
-            f.write(json.dumps(resolved_config))
+            config_cache = {
+                "_version": CONFIG_CACHE_VERSION,
+                "provider_log_info": try_get_log_state(config["provider"]),
+                "config": resolved_config
+            }
+            f.write(json.dumps(config_cache))
     return resolved_config
 
 
 def teardown_cluster(config_file, yes, workers_only, override_cluster_name,
-                     keep_min_workers):
+                     keep_min_workers, log_old_style, log_color, verbose):
     """Destroys all nodes of a Ray cluster described by a config json."""
+    cli_logger.old_style = log_old_style
+    cli_logger.color_mode = log_color
+    cli_logger.verbosity = verbose
+    cli_logger.dump_command_output = verbose == 3  # todo: add a separate flag?
 
```

Comment:
cli_logger.configure(...)

Reply:
I don't really like that option because `cli_logger` has sane defaults for all of these options and it's cleaner to set them separately (especially since you don't always want to set every single one of these). If you had something in mind for why a separate function is better, please let me know, I'm not committed to the current API.

The one reason I can see why having a function would be better is using something like `cli_logger.configure_from_args(...)` but click does not provide a raw `args` dict (afaik).

Reply:
sounds good